examples/repo_example/succinct_hash_bin_delegations.py

```diff
@@ -0,0 +1,226 @@
+difference between succinct and standard hash bin delegation read:
+https://github.com/theupdateframework/taps/blob/master/tap15.md
+
+NOTE: Metadata files will be written to a 'tmp*'-directory in CWD.
+"""
+import math
+import os
+import tempfile
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+from securesystemslib.keys import generate_ed25519_key
+from securesystemslib.signer import SSlibSigner
+
+from tuf.api.metadata import (
+    Delegations,
+    Key,
+    Metadata,
+    SuccinctRoles,
+    TargetFile,
+    Targets,
+)
+from tuf.api.serialization.json import JSONSerializer
+
+# Succinct hash bin delegation
+# ===================
+# As mentioned above, succinct hash bin delegation has the same purpose as
+# standard hash bin delegation - distribute a large number of target files over
+# multiple delegated targets metadata roles. The consequence is a smaller
+# metadata files and thus a lower network overhead for repository-client
+# communication.
+#
+# The assignment of target files to a target's metadata is done automatically,
+# based on the byte digest of the target file name. More precisely, only a
+# prefix of the target file name byte digest is needed to assign it to the
+# correct hash bin.
+#
+# In succinct hash bin delegation all delegated roles (bins) are represented by
+# one single "succinct_roles" dictionary or when a targets file is loaded with a
+# SuccinctRoles instance from tuf/api/metadata.py
+#
+# The number of bins, name prefix for all bins and key threshold are all
+# attributes that need to be configured.
+
+# Number of bins, bit length and bin number computation
+# -----------------
+# The right number of bins depends on the expected number of target files in a
+# repository. The maximum number of bins is 2 of the power of 32.
+# For the purpose of this example we choose:
+NUMBER_OF_BINS = 32
+#
+# The bit length is the number of bits that will be used to calculate the
+# corresponding bin for each target path and it's directly related to the
+# a number of bins:
+BIT_LENGTH = int(math.log2(NUMBER_OF_BINS))
+#
+# To calculate the appropriate bin number for a particular target the path to
+# that target is signed using a cryptographic algorithm (like "sha256") which
+# guarantees uniform distribution. After that, the first BIT_LENGTH of bits from
+# the byte digest are used to calculate the bin number in hexadecimal format.
+# For more information read SuccinctRoles implementation in tuf/api/metadata.py
+
+# Delegated role (bin) name format
+# -----------------
+# Each bin has a name in the format of f"{NAME_PREFIX}-{bin_number}".
+#
+# Name prefix is the common prefix of all delegated target roles (bins) in a
+# particular SuccinctRoles object. For our example it will be:
+NAME_PREFIX = "delegated_bin"
+#
+# The suffix "bin_number" is a zero-padded hexidentical number of that
+# particular bin.
```

Comment:
I suggest drastically shorten the comments until here. There are plenty of resources to refer to,
- hashed_bin_delegation.py
- `SuccinctRoles` code documentation
- TAP 15.

(The reason why `hashed_bin_delegation.py` is so verbosely commented is that none of these resources were available when I created the example script)